tsconfig.json

```diff
@@ -1,29 +1,29 @@
 {
-    "compilerOptions": {
-        "target": "es6",
-        "module": "commonjs",
-        "moduleResolution": "node",
-        "isolatedModules": false,
-        "noImplicitAny": true,
-        "noUnusedLocals": true,
-        "noUnusedParameters": true,
-        "noImplicitThis": true,
-        "strictNullChecks": true,
-        "removeComments": false,
-        "preserveConstEnums": true,
-        "suppressImplicitAnyIndexErrors": true,
-        "outDir": "./lib",
-        "declaration": true,
-        "sourceMap": true,
-        "pretty": true,
-        "lib": [
-            "es2017"
-        ]
-    },
-    "include": [
-        "src/**/*.ts"
-    ],
-    "exclude": [
-        "node_modules"
+  "compilerOptions": {
+    "target": "es6",
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "isolatedModules": false,
+    "noImplicitAny": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitThis": true,
+    "strictNullChecks": true,
+    "removeComments": false,
+    "preserveConstEnums": true,
+    "suppressImplicitAnyIndexErrors": true,
+    "outDir": "./lib",
+    "declaration": true,
+    "sourceMap": true,
+    "pretty": true,
+    "lib": [
+      "esnext"
```

Comment:
question: why switch to esnext here?

Reply:
Switched back.

An earlier version of this used async iterators, but they ended up adding a bunch of complexity that didn't pay out.